File: src/args.ts
export function hasValue(value: unknown): boolean {
  return value !== undefined && value !== null;
}

export function requireAtLeastOne(
  values: Record<string, unknown>,
  message: string
): void {
  const ok = Object.values(values).some((value) => hasValue(value));
  if (!ok) {
    throw new Error(message);
  }
}

export function requireExactlyOne(
  values: Record<string, unknown>,
  message: string
): void {
  const count = Object.values(values).filter((value) => hasValue(value)).length;
  if (count !== 1) {
    throw new Error(message);
  }
}

export function requireAll(
  values: Record<string, unknown>,
  message: string
): void {
  const ok = Object.values(values).every((value) => hasValue(value));
  if (!ok) {
    throw new Error(message);
  }
}

export function parseIntOption(value: unknown, label: string): number {
  const parsed =
    typeof value === "number" ? value : Number.parseInt(String(value), 10);
  if (Number.isNaN(parsed)) {
    throw new Error(`Invalid ${label}`);
  }
  return parsed;
}

File: pkg/args/args.go
package args

import (
	"fmt"
	"strconv"
)

func ParseIntOption(value, label string) (int, error) {
	n, err := strconv.Atoi(value)
	if err != nil {
		return 0, fmt.Errorf("%s must be an integer", label)
	}
	return n, nil
}

func RequireExactlyOne(values map[string]bool, message string) error {
	count := 0
	for _, present := range values {
		if present {
			count++
		}
	}
	if count != 1 {
		return fmt.Errorf("%s", message)
	}
	return nil
}
